src/veins/base/toolbox/Signal.cc

```diff
@@ -0,0 +1,995 @@
+//
+// Copyright (C) 2018 Fabian Bronner <fabian.bronner@ccs-labs.org>
+//
+// Documentation for these modules is at http://veins.car2x.org/
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation; either version 2 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+//
+
+//TODO: Use fixed size std::vector instead of c-style array
+
+#include "veins/base/toolbox/Signal.h"
+#include "veins/base/phyLayer/AnalogueModel.h"
+
+namespace Veins {
+
+Signal::Signal():
+	spectrum(0), values(0), numAbsoluteValues(0), numRelativeValues(0), numDataValues(0), relativeOffset(0), dataOffset(0), centerFrequencyIndex(0),
+	timingUsed(false), sendingStart(0), duration(0), propagationDelay(0),
+	analogueModelList(0), numAnalogueModelsApplied(0), senderPos(0, 0), receiverPos(0, 0),
+	bitrate(0),
+	senderModuleID(-1), senderFromGateID(-1), receiverModuleID(-1), receiverToGateID(-1)
+{
+	//
+}
+
+Signal::Signal(const Signal& other):
```

Comment:
When `Signal::values` is changed to a `std::vector` or something similar, the default copy ctor can be used instead of this.